src/test/java/ua/utility/kfsdbupgrade/IndexScriptExecutionTest.java

```diff
@@ -0,0 +1,110 @@
+ * between the SQL and pre-existing DB index. This is done through a series of
+ * queries that will line up column names from the DML against a table's actual
+ * indexed columns.
+ *
+ * If the column count or column names do not line up between the two source indexes,
+ * it is not a match, and the next table's index is then tested untill the index list
+ * on the table is exhausted.
+ *
+ * If the columns do line up when going through a tables indexes, then it is a match,
+ * even regardless of the index name, and the index building is skipped since it
+ * already exists.
+ *
+ * If ALL of a table's indexes fail to match the DML, only then is the DML index
+ * applied. It was this case that sparked this junit, as the detection was failing
+ * at some point, but no longer appears to be.
+ *
+ */
+public class IndexScriptExecutionTest extends AppTestBase {
+    private static final String SQL_INDEX_TEST_FILE = "past_problem_indexes.sql";
+
+
+    /*
+     * The logic behind the App#createExistingIndexes() method is now smart enough not
+     * to care about index names, as it does a comparison on the actual columns that
+     * are involved, and also considers the 'UNIQUE' keyword of the DML. Thus, the
+     * test file from here processes without issue. Under the hood, if the index is
+     * detected as existing, then it will be skipped for execution, and moves on to the
+     * next line in the file.
+     */
+    @Test
+    public void testProblemIndexDml() {
+        App app = getApp();
+        File file = getTestResource(SQL_INDEX_TEST_FILE);
+        boolean success = executIndexDml(app, file);
+        Assert.assertTrue("Former issue-cases did NOT process without error.", success);
+    }
+
+
+    /*
+     * Sanity check after having modified some small things in App.java, plus
+     * can be used as a general smoke test for new index DML.
+     */
+    @Test
+    public void testFullIndexDml() {
+        App app = getApp();
+        File file = new File(app.getPostUpgradeDirectory() + File.separator + App.KFS_INDEXES_SQL_PATH);
+        boolean success = executIndexDml(app, file);
+        Assert.assertTrue("Full index SQL file did NOT process without error.", success);
+    }
+
+
+    private boolean executIndexDml(App app, File file) {
+        boolean success = true;
+        deleteBiMaterializedViews(app);
+
+        Connection conn = null;
+        Statement stmt = null;
+        try {
+            conn = app.getUpgradeConnection();
+            stmt = conn.createStatement();
+
+            // This is what we're really interested in for this junit class
+            success = app.createExistingIndexes(conn, stmt, file);
+
+        } catch (Exception e) {
+            throw new RuntimeException(e);
+        } finally {
+            app.closeDbObjects(conn, stmt, null);
+        }
+
+        return success;
+    }
+
+
+    private void deleteBiMaterializedViews(App app) {
```

Comment:
It _is_ in App.java, but the test skirts it, in order to execute only the chosen index file (ie, we _don't_ want to run the full upgrade for one specialized test). And those materialized views I'd argue should not have been carried over by the DB copy job from v3, but MEM has those views under the same schema owner as ours for convenience. Those views are not used by our apps whatsoever, only a BI background process. Moreover, this is only a test, so should never be executed against a DB we care anything about.